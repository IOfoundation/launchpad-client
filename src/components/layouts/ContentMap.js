import React, {Component} from 'react';
import {PropTypes} from 'prop-types';
import {MdSearch} from 'react-icons/lib/md';
import MapView from '../map-view/Main';
import ResultPage from '../businesses/mobile/ResultPage';

class ContentMap extends Component {
  _renderReduceButton() {
    return (
      <div className="row between-xs middle-xs reducedMapBottom">
        <button
          className="btn-link btn-link-primary text-bold"
          onClick={this.props.reduceMap}
        >
          {'Reduce Map'}
        </button>
        <button
          className="btn-link btn-link-primary underline text-xs-margin"
          onClick={this.props.redoSearchInMap}
        >
          {'Redo Search in Map Area'}
        </button>
      </div>
    );
  }
  _renderExpandButton() {
    return (
      <div className="row between-xs middle-xs reducedMapBottom">
        <button
          className="btn-link btn-link-primary text-bold text-xs-margin p-0 m-bot-8"
          onClick={this.props.expandMap}
        >
          {'Expand Map'}
        </button>
        <button
          className="btn-link btn-link-primary underline text-xs-margin p-0"
          onClick={this.props.redoSearchInMap}
        >
          {'Redo Search in Map Area'}
        </button>
      </div>
    );
  }
  render() {
    const {locations, businessesMetadata, onBoundsChange, organizations} = this.props;
    return (
      <div className="businessesContainer">
        {this.props.topBar}
        <div className={
            businessesMetadata.totalOrganizations === '0' ? (
              'no-result-message-show'
            ) : (
              'no-result-message-hide'
            )
          }
        >
          <div className="no-result-message">
            <img src="/static-data/images/search-icon.png" />
            <p className="message desktop-devices">
              {'Sorry but nothing matched your search terms.'}
            </p>
            <p className="message desktop-devices">
              {'Please try Again with different Keywords'}
            </p>
            <div className="mobile-devices">
              <p className="message">
                {'Sorry but nothing matched your search terms. Please try Again with different Keywords'}
              </p>
            </div>
          </div>
        </div>
        <div className={organizations.length === 0 ? 'result-container-hide' : ''}>
          <ResultPage
            BusinessesList={this.props.children}
            locations={locations}
            onBoundsChange={onBoundsChange}
            TotalOrganizations={businessesMetadata.totalOrganizations}
          />
        </div>
        <div className={
            businessesMetadata.totalOrganizations === '0' ? (
              'result-container-hide desktop-devices'
            ) : (
              'grid desktop-devices'
            )
          }
        >
          <div
            className={
              (this.props.expanded
                ? 'col-md-7 col-xs-7 businessList p-left-0'
                : 'col-md-9 col-xs-9 businessList--reduced p-left-0') + ' list'
            }
          >
            {this.props.children}
          </div>
          <div
            className={
              'map ' +
              (this.props.expanded
                ? 'col-lg-5 col-md-5 col-xs-5 p-0'
                : 'col-lg-3 col-md-3 col-xs-3 p-0')
            }
          >
            <div
              className={
                this.props.expanded ? (
                  'map-container-collapse'
                ) : (
                  'map-container-expand'
                )
              }
            >
              <MapView locations={locations} onBoundsChange={onBoundsChange}/>
            </div>
            {this.props.expanded ? (
              this._renderReduceButton()
            ) : (
              this._renderExpandButton()
              )}
          </div>
        </div>
      </div>
    );
  }
}

ContentMap.propTypes = {
  businessesMetadata: PropTypes.object,
  children: PropTypes.node,
  expanded: PropTypes.bool,
  expandMap: PropTypes.func.isRequired,
  locations: PropTypes.array,
  onBoundsChange: PropTypes.func,
  redoSearchInMap: PropTypes.func.isRequired,
  reduceMap: PropTypes.func.isRequired,
  topBar: PropTypes.node,
  totalOrganizations: PropTypes.array,
};

export default ContentMap;
