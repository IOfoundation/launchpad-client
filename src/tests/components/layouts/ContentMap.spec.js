import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import ContentMap from 'components/layouts/ContentMap';
import organizationFixture from 'tests/fixtures/organization';

const createProps = props => {
  return {
    businesses: props.businesses,
    children: null,
    expanded: false,
    expandMap: jest.fn(),
    handleChangePage: jest.fn(),
    highlightOrgCard: jest.fn(),
    isMobile: false,
    locations: [],
    onBoundsChange: jest.fn(),
    redoSearchInMap: jest.fn(),
    reduceMap: jest.fn(),
    selectedOrg: -1,
    showLoading: false,
    toggleSwitch: false,
    topBar: null,
  };
};

describe('<ContentMap />', () => {
  it('renders snapshot of ContentMap', () => {
    const props = createProps({
      businesses: {
        locations: [],
        organizations: [organizationFixture, organizationFixture],
        metadata: {totalOrganization: '50'},
        appliedFilters: {
          pagination: {
            first: {page: 1, per_page: 4},
            last: {},
            next: {},
            currentPage: 1,
          },
        },
      },
    });
    const wrapper = shallow(<ContentMap {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
