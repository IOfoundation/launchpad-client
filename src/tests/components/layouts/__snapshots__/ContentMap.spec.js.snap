// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentMap /> renders snapshot of ContentMap 1`] = `
<div
  className="businessesContainer"
>
  <div
    className="map m-bot-20 col-lg-3 col-md-3 col-xs-3 p-0"
  >
    <div
      className="map-container-expand"
    >
      <Main
        expanded={false}
        highlightOrgCard={[Function]}
        isMobile={false}
        locations={Array []}
        onBoundsChange={[Function]}
        organizations={
          Array [
            Object {
              "contacts": Array [
                Object {
                  "email": "jondoe@test.org",
                  "id": 1,
                  "name": "Jon Doe",
                  "phones": Array [
                    Object {
                      "number": "(530) 795-1520",
                    },
                  ],
                },
              ],
              "description": "orgDescription",
              "email": "jondoe@something.org",
              "id": 1,
              "locations": Array [
                Object {
                  "address": Object {
                    "address_1": "123 Road St",
                    "address_2": null,
                    "city": "Sacramento",
                    "postal_code": "12345",
                    "state_province": "CA",
                  },
                  "id": 1,
                  "name": "location1",
                  "phones": Array [
                    Object {
                      "id": 2,
                      "number": "(123) 456 - 7890",
                    },
                  ],
                },
                Object {
                  "address": Object {
                    "address_1": "456 Street Rd",
                    "address_2": null,
                    "city": "Sacramento",
                    "postal_code": "12345",
                    "state_province": "CA",
                  },
                  "id": 2,
                  "name": "location2",
                  "phones": Array [
                    Object {
                      "id": 3,
                      "number": "(123) 456 - 7890",
                    },
                  ],
                },
              ],
              "name": "organization1",
              "phones": Array [
                Object {
                  "id": 1,
                  "number": "(123) 456 - 7890",
                },
              ],
              "services": Array [
                Object {
                  "categories": Array [
                    Object {
                      "id": 54,
                      "name": "category1",
                    },
                    Object {
                      "id": 55,
                      "name": "category2",
                    },
                    Object {
                      "id": 56,
                      "name": "category3",
                    },
                  ],
                  "description": "lorem ipsum...",
                  "email": null,
                  "id": 1,
                  "name": "service1",
                  "phones": Array [],
                },
                Object {
                  "categories": Array [
                    Object {
                      "id": 54,
                      "name": "category1",
                    },
                  ],
                  "description": "lorem ipsum...",
                  "email": null,
                  "id": 2,
                  "name": "service2",
                  "phones": Array [],
                },
              ],
            },
            Object {
              "contacts": Array [
                Object {
                  "email": "jondoe@test.org",
                  "id": 1,
                  "name": "Jon Doe",
                  "phones": Array [
                    Object {
                      "number": "(530) 795-1520",
                    },
                  ],
                },
              ],
              "description": "orgDescription",
              "email": "jondoe@something.org",
              "id": 1,
              "locations": Array [
                Object {
                  "address": Object {
                    "address_1": "123 Road St",
                    "address_2": null,
                    "city": "Sacramento",
                    "postal_code": "12345",
                    "state_province": "CA",
                  },
                  "id": 1,
                  "name": "location1",
                  "phones": Array [
                    Object {
                      "id": 2,
                      "number": "(123) 456 - 7890",
                    },
                  ],
                },
                Object {
                  "address": Object {
                    "address_1": "456 Street Rd",
                    "address_2": null,
                    "city": "Sacramento",
                    "postal_code": "12345",
                    "state_province": "CA",
                  },
                  "id": 2,
                  "name": "location2",
                  "phones": Array [
                    Object {
                      "id": 3,
                      "number": "(123) 456 - 7890",
                    },
                  ],
                },
              ],
              "name": "organization1",
              "phones": Array [
                Object {
                  "id": 1,
                  "number": "(123) 456 - 7890",
                },
              ],
              "services": Array [
                Object {
                  "categories": Array [
                    Object {
                      "id": 54,
                      "name": "category1",
                    },
                    Object {
                      "id": 55,
                      "name": "category2",
                    },
                    Object {
                      "id": 56,
                      "name": "category3",
                    },
                  ],
                  "description": "lorem ipsum...",
                  "email": null,
                  "id": 1,
                  "name": "service1",
                  "phones": Array [],
                },
                Object {
                  "categories": Array [
                    Object {
                      "id": 54,
                      "name": "category1",
                    },
                  ],
                  "description": "lorem ipsum...",
                  "email": null,
                  "id": 2,
                  "name": "service2",
                  "phones": Array [],
                },
              ],
            },
          ]
        }
        showLoading={false}
        toggleSwitch={false}
      />
    </div>
    <div
      className="row between-xs middle-xs reducedMapBottom"
    >
      <button
        className="expand-btn map-btn btn-link btn-link-primary text-bold text-xs-margin p-0 m-bot-8 m-top-5"
        onClick={[Function]}
      >
        Expand Map
      </button>
      <div
        className="map-btn"
      >
        <div
          className="float-right"
        >
          <button
            className="toggle-btn toggle-off"
            onClick={[Function]}
          >
            <span />
          </button>
          <button
            className="btn-link btn-link-primary underline text-xs-margin text-bold"
            onClick={[Function]}
          >
            Search in Map Area
          </button>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="result-container-hide desktop-devices"
    >
      <div
        className="col-md-12 col-xs-12 businessList--reduced p-left-0 p-right-2 list"
      />
    </div>
  </div>
</div>
`;
